File: test-fixtures/languages/php/micro/auth_service.php
<?php

function verify_token($token) {
    $conn = new mysqli("localhost", "root", "", "auth");
    $query = "SELECT user_id FROM tokens WHERE token = '" . $token . "'";
    $result = $conn->query($query); // sink: SQL injection
    if ($row = $result->fetch_assoc()) {
        return $row['user_id'];
    }
    $conn->close();
    return null;
}

function run_admin_command($user_id, $cmd) {
    if ($user_id !== null) {
        exec("notify-admin " . $cmd); // sink: command injection
    }
}

File: test-fixtures/languages/python/micro/auth_service.py
import sqlite3
import os


def verify_token(token):
    conn = sqlite3.connect("auth.db")
    cursor = conn.cursor()
    query = "SELECT user_id FROM tokens WHERE token = '" + token + "'"
    cursor.execute(query)  # sink: SQL injection
    row = cursor.fetchone()
    conn.close()
    if row:
        return row[0]
    return None


def run_admin_command(user_id, cmd):
    if user_id:
        os.system("notify-admin " + cmd)  # sink: command injection
